### PR TITLE
feat(radio-list): add component

### PR DESCRIPTION
**Pull request by pedrorebelocosta**
image

### CODE REVIEW

File: packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { JSX, h } from '@stencil/core';
+import { LINK_MATCHER_REGEX } from './radio-list-item.config';
+import { EAnchorTarget } from '../../types';
+
+export const buildDescription = (description?: string): JSX.Element[] => {
```
Comment (djcouto) [inline packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:5]: Are we going forward with this approach? Seems very error prune.
Besides, what if want some part of the description to be bold? How would we achieve that?
Seems to me that a slots solution, despite requiring more work from the developer, would fit every use case we would need.

**Review by djcouto — CHANGES_REQUESTED** [review]
I left one question. Apart from that, LGTM 👍

File: packages/ui-components/src/components/radio-list/radio-list.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	render() {
+		return (
+			<Host>
+				<div class="radio-list-container">
+					{this.label && <kv-form-label label={this.label} required={this.required} />}
+					{this.options.map(item => {
+						return (
+							<kv-radio-list-item
+								{...item}
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/radio-list/radio-list.tsx:37]: what item has more to destruct?

File: packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	if (!description) return;
+
+	const linkMatches = [...description.matchAll(LINK_MATCHER_REGEX)];
+	const parsedJSX: JSX.Element[] = [];
+	let curIdx = 0;
+
+	for (const match of linkMatches) {
+		const beforeLink = description.slice(curIdx, match.index);
+		if (beforeLink.length > 0) {
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:14]: It is a problem if we push an empty string for `parsedJSX`?

File: packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx (hunk 1 above)
Comment (miguelpinto98) [inline packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:5]: Probably we can export this util to be used in other situations like this?

**Review by miguelpinto98 — CHANGES_REQUESTED** [review]
Almost done! Only check my questions here. 🚀

File: packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx (hunk 1 above)
Comment (djcouto) [reply packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:5]: Why don't we use something like this?

https://github.com/markedjs/marked
https://github.com/probablyup/markdown-to-jsx

File: packages/ui-components/src/components/radio-list-item/radio-list-item.config.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const LINK_MATCHER_REGEX = /\[(.+?)\]\((https?:\/\/.+?)\)/g;
```
[the remark this answers is not in the document]
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio-list-item/radio-list-item.config.ts:1]: this matches both `http` and `https`, should we remove the check completely?

File: packages/ui-components/src/components/radio-list/radio-list.tsx (hunk 2 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio-list/radio-list.tsx:37]: `label` and `description`, just these 2 props since it's just 2 props, I'll make them explicit here

File: packages/ui-components/src/components/radio/radio.tsx [hunk 5]
```diff
@@ -76,7 +76,7 @@ export class KvRadio {
 					onClick={this.clickThrottler}
 				>
 					<div tabIndex={this.isChecked || this.isDisabled ? -1 : 0} class="circle">
-						<div class="checked-icon"></div>
+						{this.isChecked && <div class="checked-icon"></div>}
```
[the remark this answers is not in the document]
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio/radio.tsx:79]: You're absolutely right 👍

File: packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx (hunk 3 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:14]: as parsedJSX is initialized as an empty array, if you pass an empty string the return value will be an empty array which will not render anything

File: packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx (hunk 1 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:5]: As discussed earlier, this bit of code is still attached / used by this component only, moving forward, we'll have a separation of this parsing logic from the component.

Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio-list-item/radio-list-item.helper.tsx:5]: Thank you for the input, moving forward I'll isolate the parsing logic from this component, most likely using one of the aforementioned powerful libs!

**Review by miguelpinto98 — APPROVED** [review]
🚀

File: packages/ui-components/src/components/radio-list/radio-list.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
+@import '../../assets/styles/globals';
+
+.radio-list-container {
+	display: flex;
+	flex-direction: column;
+
+	.radio-list-item {
+		margin-bottom: $spacing-2x;
```
Comment (djcouto) [inline packages/ui-components/src/components/radio-list/radio-list.scss:8]: You can use `gap: $spacing-2x` on the `.radio-list-container` to avoid the last `.radio-list-item` having an unnecessary margin-bottom.

File: packages/ui-components/src/components/radio/radio.base.scss [hunk 7]
```diff
@@ -35,10 +35,6 @@
 
 		.circle {
 			border-color: var(--radio-input-default-color);
-
-			.checked-icon {
```
Comment (djcouto) [inline packages/ui-components/src/components/radio/radio.base.scss:39]: What's the purpose of this change?

**Review by djcouto — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/components/radio/radio.base.scss (hunk 7 above)
Comment (miguelpinto98) [reply packages/ui-components/src/components/radio/radio.base.scss:39]: already checked with @djcouto

File: packages/ui-components/src/components/radio-list/radio-list.scss (hunk 6 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/radio-list/radio-list.scss:8]: that would cause a 4px difference since form-label already has 4px margin bottom, but i'll change it to something simpler :)

**Review by djcouto — APPROVED** [review]
